include/gsl/gsl-lite.hpp

```diff
@@ -1439,6 +1448,17 @@ class not_null
 #else
 # define gsl_not_null_explicit /*explicit*/
 #endif
+#if gsl_HAVE( FUNCTION_REF_QUALIFIER )
+# define gsl_not_null_LVALUE_REF &
+# define gsl_not_null_CONVERSION_REF
+#else
+# define gsl_not_null_LVALUE_REF
+# if gsl_CONFIG( NOT_NULL_GET_BY_CONST_REF )
+#  define gsl_not_null_CONVERSION_REF const &
+# else
+#  define gsl_not_null_CONVERSION_REF
+# endif
+#endif
```

Comment:
These should be moved closer to their uses to make understanding the code easier.

Reply:
So should `gsl_not_null_explicit`, I suppose? I thought I'd keep the macro definitions together as a group... but yes, I can move them closer.